src/main.rs

```diff
@@ -0,0 +1,107 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, NodeIndex},
+    Felt, Word, ONE,
+    {hash::rpo::Rpo256, merkle::TieredSmt},
+};
+use std::time::Instant;
+
+fn main() {
+    construction(1000).unwrap();
+    construction(1_000_000).unwrap();
+
+    insertion(1000).unwrap();
+    insertion(1_000_000).unwrap();
+
+    proof_generation(1000).unwrap();
+    proof_generation(1_000_000).unwrap();
+}
```

Comment:
We should read the size of the tree from the command line so that we can run benchmarks for a tree of any size. We could use `clap` for this (see https://github.com/0xPolygonMiden/miden-vm/pull/1032).